hw/ip/rv_dm/data/rv_dm_testplan.hjson

```diff
@@ -463,6 +463,22 @@
       stage: V1
       tests: ["rv_dm_halt_resume_whereto"]
     }
+    {
+      name: abstractcmd_status
+      desc: '''
+            Verify the functionality of abstract commands by reading memory interface abstract
+            command registers.
+
+            - Program jtag_dmi_ral.dmcontrol.haltreq to 1'b1.
+            - Acknowledge haltreq by setting tl_mem_ral.halted to 0.
```

Comment:
If I understand correctly, we expect that `rv_dm` will set a `halted` register to the ID of the core that is halted. This ID is not zero. Do you mean something like this?
```
  - See that tl_mem_ral.halted becomes the core IDCODE, showing the halt request has been acknowledged.
```

I left a less definite note on a [previous version](https://github.com/lowRISC/opentitan/pull/21381#discussion_r1490918576).

Reply:
Yes, in debug mode hart write its ID to this address to acknowledge that it is halted.

Reply:
Great. In which case, the testplan should probably talk about *reading* that register (and hopefully seeing the hart's id).

Reply:
But we do not have hart in our case. so we have to write on halted register on behalf of hart to indicate that hart is halted. isn't it?

Reply:
If you're not sure how this should work, I'd suggest you have a look at the top-level ports in `dm_top.sv`, or the "*halted*" signals in `dm_mem.sv`.

Thinking about it, we could skip talking about the halted register entirely, by looking at the `debug_req_o` signal from `rv_dm`. This appears in `rv_dm_if`.

The whole point of this sort of document is to figure out what we're doing *before* typing any code. I've reviewed your proposed implementation and it works by writing to a read-only register. Clearly wrong, but already obvious at the planning stage. We could have figured things out before typing any verilog.

Reply:
I've just looked at #21543. And I'd got completely the wrong end of the stick about how this works, sorry!

I think you're right, and that we probably do need to tell the debug module that we've halted. I wonder whether we should also add an entry to this list mentioning that we expect to see the debug module halt the hart.

Reply:
After writing on halted register we can check status of debug module by checking jtag_dmi_ral.dmstatus.anyhalted/allhalted bit to see that debug module tells that hart is halted. we can include that check in testplan. but i think it is ok otherwise as well.